src/accelerate/state.py

```diff
@@ -778,16 +778,11 @@ def set_device(self):
         else:
             if device == "gpu":
                 device = "cuda"
-            self.device = torch.device(device, self.local_process_index)
-        if self.device is not None:
-            if device == "xpu":
-                torch.xpu.set_device(self.device)
-            elif device == "mlu":
-                torch.mlu.set_device(self.device)
-            elif device == "npu":
-                torch.npu.set_device(self.device)
-            elif device == "cuda":
-                torch.cuda.set_device(self.device)
+            assert device in ("cpu", "cuda", "mlu", "npu", "xpu")
```

Comment:
We generally don't do hard asserts like this here, and as Benjamin pointed out we should be covered for this.
```suggestion
```